synthetic_example/SyntheticDataset.py

```diff
@@ -0,0 +1,141 @@
+            )
+            if os.path.exists(zarr_path):
+                zarr_paths.append(zarr_path)
+            else:
+                print(f"Data for year {year} not found!!!")
+        ds = xr.open_mfdataset(
+            zarr_paths, engine="zarr", consolidated=True, combine="by_coords"
+        )[self.input_vars]
+        self.length = ds.sizes["time"]
+        return ds
+
+    def __len__(self):
+        """Returns the total number of samples in the dataset."""
+        return self.length
+
+    def fetch_timeseries(self, forecast_step=1):
+        """
+        Fetches the input and target timeseries data for a given forecast step.
+        """
+        ds_x = self.dataset.isel(time=slice(None, -forecast_step))
+        ds_y = self.dataset.isel(time=slice(forecast_step, None))
+        return ds_x, ds_y
+
+    def normalize(self, mean_file=None, std_file=None):
+        """
+        Normalize the dataset using the mean and std files.
+        """
+        if mean_file is not None and std_file is not None:
+            mean = xr.open_dataset(mean_file)
+            std = xr.open_dataset(std_file)
+        else:
+            mean = self.dataset.mean(dim="time")
+            std = self.dataset.std(dim="time")
+        self.dataset = (self.dataset - mean) / std
+        self.normalized = True
+
+    def __repr__(self):
+        """Returns a summary of all datasets loaded."""
+        return self.dataset.__repr__()
+
+    def __getitem__(self, index):
+        """Enable direct indexing"""
+        x_data = np.zeros([6, 640, 1280], dtype=np.float32)
+        y_data = np.zeros([6, 640, 1280], dtype=np.float32)
+        return (x_data, y_data)
+
+
+class PyTorchERA5Dataset(Dataset):
+    """
+    Wraps the ERA5TimeSeriesDataset so it can be used in PyTorch DataLoader.
+    """
+
+    def __init__(self, era5_dataset, forecast_step=1):
+        """
+        era5_dataset (ERA5Dataset): An instance of the custom ERA5 dataset.
+        forecast_step (int): The forecast step to use for fetching timeseries data.
+        """
+        self.era5_dataset = era5_dataset
+        self.forecast_step = forecast_step
+        self.ds_x, self.ds_y = self.era5_dataset.fetch_timeseries(
+            forecast_step=self.forecast_step
+        )
+
+    def __len__(self):
+        """Returns the total number of samples in the dataset."""
+        return self.ds_x.sizes["time"]
+
+    def __getitem__(self, index):
+        """
+        Returns a single sample (input, target) as PyTorch tensors.
+
+        Args:
+            index (int): Index of the sample to retrieve.
+
+        Returns:
+            tuple: (input_tensor, target_tensor)
+        """
+        # Extract data at the given index
+        x_tensor = torch.zeros([6, 640, 1280], dtype=torch.float32)
+        # TODO: I'm surprised this is [6, 640, 1280] rather than [1, 640, 1280]
+        y_tensor = torch.zeros([6, 640, 1280], dtype=torch.float32)
```

Comment:
@negin513 @weiji14 Can you confirm that this tensor shape is expected? I would've thought we'd only have length 1 on the first dimension since we only have one target variable?